compiler/backend/cgen.nim

```diff
@@ -1063,6 +1063,18 @@ proc genProcAux(m: BModule, prc: PSym) =
         #incl(res.loc.flags, lfIndirect)
         res.loc.storage = OnUnknown
 
+  # for now, we treat all compilerprocs as being able to run in a boot
+  # environment where the error flag is not yet accessible. This is not quite
+  # correct, and a dedicated facility for designating runtime procedures as
+  # usable in a boot environment is eventually required
```

Comment:
One thing I can think of is that we check to see if some symbol (e.g. `system.bootRTErrorFlag`) is defined, that symbol acts as a sentinel indicating that we're far enough along in runtime environment setup that exceptions can now be used. When we analyse any code relying on try/except/raise, has raise in its effects, etc... we check for the symbol and if not present can handle it later in codegen and or raise it as a sem error as appropriate. I'm sure I've missed an important detail, but that's what currently comes to mind.

Writing it out and thinking about it a little, I'm not sure if the above is a contender for temporary solution, I suspect plenty of rearrange of `system` and related modules would be required in order to get this working.

Reply:
Yeah, I think checking for the presence of a symbol or for a procedure's owner is the way to go.

As you noted, the `system` has to be reordered quite a bit for that to work. That has to happen eventually anyway, but at this point, I'd wait with these efforts until the `system` and related modules are no longer shared between the normal compiler and csources compiler, as we don't need to consider compatibility as much then.